### PR TITLE
Feat/historical index

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**
Added functions for getting historical rates (fixed & variable)
- functions are added as getters in AMM (following the stateless SDK format)
- they return an array of datapoints and allow for setting granularity & timeframe

### CODE REVIEW

File: src/entities/amm/amm.ts [hunk 1]
```diff
@@ -1482,6 +1483,36 @@ export class AMM {
...
+    return result;
+  }
+
+  public async getHistoricalFixedRate(
+    subgraphUrl: string,
+    filters: {
+      granularity: Granularity;
+      timeframeMs: number;
+    },
+  ): Promise<any[]> {
```
Comment (avniculae) [inline src/entities/amm/amm.ts:1511]: `Promise<HistoricalRates[]>` is better

File: src/entities/amm/amm.ts [hunk 2]
```diff
@@ -1482,6 +1483,36 @@ export class AMM {
...
 
+  // historocal rates
+
+  public async getHistoricalVariableRate(
+    subgraphUrl: string,
+    filters: {
+      granularity: Granularity;
+      timeframeMs: number;
+    },
+  ): Promise<any[]> {
```
Comment (avniculae) [inline src/entities/amm/amm.ts:1494]: `Promise<HistoricalRates[]>` is better

File: src/entities/amm/amm.ts [hunk 3]
```diff
@@ -1482,6 +1483,36 @@ export class AMM {
     return tickToFixedRate(closestUsableTick).toNumber();
   }
 
+  // historocal rates
+
+  public async getHistoricalVariableRate(
+    subgraphUrl: string,
```
Comment (avniculae) [inline src/entities/amm/amm.ts:1489]: replace with `chainId` and use `getSubgraphURL(chainId, SubgraphURLEnum.voltzProtocol)` to get the subgraph URL

File: src/entities/amm/amm.ts [hunk 4]
```diff
@@ -1482,6 +1483,36 @@ export class AMM {
...
+      false,
+      filters,
+      undefined,
+      this.rateOracle.id,
+    );
+    return result;
+  }
+
+  public async getHistoricalFixedRate(
+    subgraphUrl: string,
```
Comment (avniculae) [inline src/entities/amm/amm.ts:1506]: replace with `chainId` and use `getSubgraphURL(chainId, SubgraphURLEnum.voltzProtocol)` to get the subgraph URL

File: src/entities/amm/amm.ts (hunk 1 above)
Comment (ioanaroceanu2000) [reply src/entities/amm/amm.ts]: [13c3037](https://github.com/Voltz-Protocol/v1-sdk/pull/206/commits/13c30377bd06863b9fa0ffd0afeae77c4d4abda3)

File: src/entities/amm/amm.ts (hunk 3 above)
Comment (ioanaroceanu2000) [reply src/entities/amm/amm.ts]: ah nicee. I implemented that [13c3037](https://github.com/Voltz-Protocol/v1-sdk/pull/206/commits/13c30377bd06863b9fa0ffd0afeae77c4d4abda3)

**Review by avniculae — APPROVED** [review]

**Pull request merged**